File: CSES/DynamicProgramming/CoinCombinationI.cpp
#include <iostream>
#include <vector>

using namespace std;

const int MOD = 1000000007;

int main() {
    int n, x;
    cin >> n >> x;

    vector<int> coins(n);
    for (int i = 0; i < n; i++) {
        cin >> coins[i];
    }

    vector<int> dp(x + 1, 0);
    dp[0] = 1;

    for (int i = 1; i <= x; i++) {
        for (int coin : coins) {
            if (i >= coin) {
                dp[i] = (dp[i] + dp[i - coin]) % MOD;
            }
        }
    }

    cout << dp[x] << endl;

    return 0;
}

File: CSES/DynamicProgramming/CoinCombinationI.py
n,x = tuple(map(int,input().split()))
coins = list(map(int,input().split()))
mod = 10**9+7
dp = [0]*(x+1)
dp[0] = 1

for i in range(1,x+1):
    for coin in coins:
        if i>=coin:
            dp[i] = (dp[i]+dp[i-coin])%mod

print(dp[-1])